gui/src/app/windows/LoadProjectWindow/LoadProjectPanel.tsx

```diff
@@ -119,25 +135,43 @@ const LoadProjectPanel: FunctionComponent<LoadProjectProps> = ({ onClose }) => {
     [importUploadedZip],
   );
 
+  const { urlToLoad, setUrlToLoad } = useUrlLoader({ onClose, setErrorText });
+
   return (
     <div className="dialogWrapper">
       <Stack spacing={2}>
-        <div>
-          You can upload:
-          <ul>
-            <li>A .zip file that was previously exported</li>
-            <li>
-              A directory of files that were extracted from an exported .zip
-              file
-            </li>
-            <li>An individual *.stan file</li>
-            <li>
-              Other individual project files (data.json, meta.json, data.py,
-              etc.)
-            </li>
-          </ul>
-        </div>
-        <UploadFiles height={300} onUpload={onUpload} />
+        <FormControl margin="normal">
+          <TextField
+            variant="standard"
+            label="Project URL"
+            value={urlToLoad}
+            onChange={(e) => setUrlToLoad(e.target.value.trim())}
```

Comment:
I'm not sure having this automatically execute the redirect is the right approach. It's nice to get the immediate feedback on whether the target URL is valid, but I can imagine a situation where I actually want to edit the string manually or something.

Not the biggest deal in the world, but I think I'd prefer if the callback that does the navigation were separate, and triggered on the text element losing focus (or just "user hit enter").

Reply:
I'd be happy to do whatever, the instantaneous nature is shared with if you drop a zip file in the box below but I guess that feels more final.

Do you know what the callback you're suggestion would be called? There are dozens of `onSomething` props available for this element and I don't know which does what you describes

Reply:
If you want something to execute when an element loses focus, the relevant callback is `onBlur()`. For catching the `enter` stroke, it'd be `onKeyPress()`.